doc-surrealdb_versioned_docs/version-2.x/reference-guide/security-best-practices.mdx

```diff
@@ -0,0 +1,323 @@
+
+In cases where SurrealDB is publicly exposed in environments where any sort of information leakage is unacceptable, the `--no-identification-headers` flag can be enabled, which will result in the SurrealDB server no longer responding to HTTP requests with headers that identify the product or its current version to prevent passive fingerprinting and metadata indexing. Note that this will not prevent active fingerprinting such as directly querying the `/version` endpoint if available or directly attempting to exploit a known security vulnerability without regard for compatibility. On the other hand, consider whether or not enabling this feature is compatible with your clients, which may rely on these headers in order to identify the version of SurrealDB running on the server.
+
+**Example**
+
+```bash
+# When launched without "--no-identification-headers":
+$ curl -vvv "127.0.0.1:8000"
+*   Trying 127.0.0.1:8000...
+* Connected to 127.0.0.1 (127.0.0.1) port 8000 (#0)
+> GET / HTTP/1.1
+> Host: 127.0.0.1:8000
+> User-Agent: curl/7.81.0
+> Accept: */*
+>
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 307 Temporary Redirect
+< location: https://surrealdb.com/app
+< access-control-allow-origin: *
+< vary: origin
+< vary: access-control-request-method
+< vary: access-control-request-headers
+# highlight-start
+< surreal-version: surrealdb-2.0.0+20240612.2184e80f
+< server: SurrealDB
+# highlight-end
+< x-request-id: 157413ce-7cc4-41a1-a93b-0940bf87874c
+< content-length: 0
+< date: Mon, 17 Jun 2024 15:47:29 GMT
+<
+* Connection #0 to host 127.0.0.1 left intact
+
+# When launched with "--no-identification-headers":
+$ curl -vvv "127.0.0.1:8000"
+*   Trying 127.0.0.1:8000...
+* Connected to 127.0.0.1 (127.0.0.1) port 8000 (#0)
+> GET / HTTP/1.1
+> Host: 127.0.0.1:8000
+> User-Agent: curl/7.81.0
+> Accept: */*
+>
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 307 Temporary Redirect
+< location: https://surrealdb.com/app
+< access-control-allow-origin: *
+< vary: origin
+< vary: access-control-request-method
+< vary: access-control-request-headers
+< x-request-id: deec3301-e930-4389-a0da-b2a336bd2631
+< content-length: 0
+< date: Mon, 17 Jun 2024 15:49:43 GMT
+<
+* Connection #0 to host 127.0.0.1 left intact
+```
+
+## Least Privilege
+
+When defining [system users](https://surrealdb.com/docs/surrealdb/2.x/security/authentication#system-users) in SurrealDB, you may assign them [roles](https://surrealdb.com/docs/surrealdb/2.x/surrealql/statements/define/user#roles) that will limit the actions they can perform inside the level where they are defined. Ensure that you employ the principle of least privilege and create users at the lowest level possible and with the minimum role in order to be able to perform their duties inside of SurrealDB. This will mitigate some of the risk in the case where credentials for that user are ever compromised.
+
+**Examples**
```

Comment:
```suggestion
### Example: Define users with least privilege
```

Reply:
Probably not "least privilege" as that will depend on the use case. I would instead say "Define users with specific roles".